agent/agent_pool.go

```diff
@@ -64,11 +93,81 @@ func (r *AgentPool) runWorker(ctx context.Context, worker *AgentWorker, im *Idle
 	defer worker.Disconnect(ctx)
 
 	// Starts the agent worker and wait for it to finish.
-	return worker.Start(ctx, im)
+	return worker.Start(ctx, r.idleMonitor)
 }
 
 func (r *AgentPool) Stop(graceful bool) {
 	for _, worker := range r.workers {
 		worker.Stop(graceful)
 	}
 }
+
+func agentWorkerHealthHandler(aw *AgentWorker) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		aw.stats.Lock()
+		defer aw.stats.Unlock()
+
+		if aw.stats.lastHeartbeatError != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "ERROR: last heartbeat failed: %v. last successful was %v ago", aw.stats.lastHeartbeatError, time.Since(aw.stats.lastHeartbeat))
+		} else {
+			if aw.stats.lastHeartbeat.IsZero() {
+				fmt.Fprintf(w, "OK: no heartbeat yet")
+			} else {
+				fmt.Fprintf(w, "OK: last heartbeat successful %v ago", time.Since(aw.stats.lastHeartbeat))
+			}
+		}
+	}
+}
+
+func (ap *AgentPool) statusJSONHandler(l logger.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		type agentWorkerStatus struct {
+			Status       agentWorkerState `json:"status"`
+			CurrentJobID string           `json:"current_job_id,omitempty"`
+			ID           string           `json:"id"`
```

Comment:
Noting these are both uuids – did you check what the pattern (naming the attribute id or uuid) was elsewhere in the agent? and in env vars etc?

Reply:
we generally call these IDs in the agent, even though they're almost always UUIDs - i suspect the logic is that all UUIDs are IDs, and the scheme might change on the buildkite end at some point? regardless, i suspect the fact that these are UUIDs (rather than some other ID scheme) is of little interest to the users of this endpoint